### PR TITLE
add for admin the user filter by role functionality

### LINKED ISSUE #10: 009-feature-users-filter-by-role
On admin-users page when selecting the filter with admin tag, just admins should be displayed. When selecting users, just users should be displayed

### PR DESCRIPTION
**Pull request by ApostolaT**
The admin should be able to filter the users by their role.
If a role is selected, the pages must be for the users of the selected type.
On changing the page, the role selection should remain the same

### CODE REVIEW

File: src/Controllers/UserController.php [hunk 1]
```diff
@@ -178,23 +165,96 @@ public function updateUser(RequestInterface $request)
...
+        if ($role === 'admin') {
+            return $this->userService->selectAllLikeFromPage($role, $currentPage);
+        }
 
+        return $this->userService->selectAllLikeFromPage('user', $currentPage);
+    }
     /**
      * This function is called to create a paginator for
-     * all users page.
+     * all users page based on role. If roll is different from
```
Comment (ApostolaT) [inline src/Controllers/UserController.php]: Change roll to role

File: src/Repositories/UserRepository.php [hunk 2]
```diff
@@ -1,11 +1,69 @@
...
+            "SELECT * FROM $tableName WHERE role like :role LIMIT :limit OFFSET :offset";
+        $query = $this->pdo->prepare($queryString);
+        $query->bindParam(':role', $role);
+        $query->bindParam(':limit', $limit);
+        $query->bindParam(':offset', $offset);
+        $query->execute();
 
+        return $query->fetchAll();
+    }
 }
```
Comment (ApostolaT) [inline src/Repositories/UserRepository.php]: Add new line

File: src/Controllers/UserController.php [hunk 3]
```diff
@@ -26,32 +25,22 @@ public function setService(AbstractService $userService)
...
         $this->userService = $userService;
     }
+
     /**
      * This function if called by an admin displays the current page
      * with all the users from the system. If a user calls to see all
      * the users, he is redirected to a login page;
      * @param RequestInterface $request
      * @return \Framework\Http\Message|Response|\Psr\Http\Message\MessageInterface
+     * @throws \Exception
```
Comment (noonlit) [inline src/Controllers/UserController.php]: who catches it? :)

File: src/Controllers/UserController.php [hunk 4]
```diff
@@ -178,23 +165,96 @@ public function updateUser(RequestInterface $request)
...
+    }
+    /**
+     * This function extracts from Repository the user entities calling
+     * the selectAllLikeFromPage function associated to the userService,
+     * passing the role value and the page value
+     * @param string $role
+     * @param int $currentPage
+     * @return array|null
+     * @throws \Exception
+     */
```
Comment (victortodoran) [inline src/Controllers/UserController.php]: Extremely weird. Let us discuss this pls.

File: src/Controllers/UserController.php [hunk 5]
```diff
@@ -178,23 +165,96 @@ public function updateUser(RequestInterface $request)
...
+    private function getRenderParamsBasedOnRoleAndPage(string $role, int $currentPage): ?array
```
Comment (ApostolaT) [inline src/Controllers/UserController.php]: Delete this function

**Pull request merged**